core/src/test/scala/unit/kafka/controller/ControllerEventManagerTest.scala

```diff
@@ -135,6 +135,43 @@ class ControllerEventManagerTest {
     assertEquals(500, queueTimeHistogram.max, 0.01)
   }
 
+  @Test
+  def testEventQueueTimeResetOnTimeout(): Unit = {
+    val metricName = "kafka.controller:type=ControllerEventManager,name=EventQueueTimeMs"
+    val controllerStats = new ControllerStats
+    val time = new MockTime()
+    val latch = new CountDownLatch(1)
+    val processedEvents = new AtomicInteger()
+
+    val eventProcessor = new ControllerEventProcessor {
+      override def process(event: ControllerEvent): Unit = {
+        latch.await()
+        processedEvents.incrementAndGet()
+      }
+      override def preempt(event: ControllerEvent): Unit = {}
+    }
+
+    controllerEventManager = new ControllerEventManager(0, eventProcessor,
+      time, controllerStats.rateAndTimeMetrics, 1)
+    controllerEventManager.start()
+
+    controllerEventManager.put(TopicChange)
+    controllerEventManager.put(TopicChange)
+    latch.countDown()
```

Comment:
Why is this `CountdownLatch` necessary? Seems it’s just delaying the wait condition being true - `processedEvents.get() == 2`